TestCloseConnectionWhenServerPrefaceNotReceived flaky

```
go test -race -cpu 1,4 -timeout 7m google.golang.org/grpc/...
E1218 22:40:53.010860    5280 pickfirst.go:58] pickfirstBalancer: failed to NewSubConn: grpc: cannot create SubConn with empty address list
==================
WARNING: DATA RACE
Write at 0x000000e17608 by goroutine 140:
  google.golang.org/grpc.TestCloseConnectionWhenServerPrefaceNotReceived.func1()
      /home/travis/gopath/src/google.golang.org/grpc/clientconn_test.go:175 +0x3a
  google.golang.org/grpc.TestCloseConnectionWhenServerPrefaceNotReceived()
      /home/travis/gopath/src/google.golang.org/grpc/clientconn_test.go:253 +0x497
  testing.tRunner()
      /home/travis/.gimme/versions/go1.9.2.linux.amd64/src/testing/testing.go:746 +0x16c
Previous read at 0x000000e17608 by goroutine 201:
  google.golang.org/grpc.(*addrConn).resetTransport()
      /home/travis/gopath/src/google.golang.org/grpc/clientconn.go:1052 +0x3ec
  google.golang.org/grpc.(*addrConn).transportMonitor()
      /home/travis/gopath/src/google.golang.org/grpc/clientconn.go:1251 +0x619
  google.golang.org/grpc.(*addrConn).connect.func1()
      /home/travis/gopath/src/google.golang.org/grpc/clientconn.go:821 +0x268
Goroutine 140 (running) created at:
  testing.(*T).Run()
      /home/travis/.gimme/versions/go1.9.2.linux.amd64/src/testing/testing.go:789 +0x568
  testing.runTests.func1()
      /home/travis/.gimme/versions/go1.9.2.linux.amd64/src/testing/testing.go:1004 +0xa7
  testing.tRunner()
      /home/travis/.gimme/versions/go1.9.2.linux.amd64/src/testing/testing.go:746 +0x16c
  testing.runTests()
      /home/travis/.gimme/versions/go1.9.2.linux.amd64/src/testing/testing.go:1002 +0x521
  testing.(*M).Run()
      /home/travis/.gimme/versions/go1.9.2.linux.amd64/src/testing/testing.go:921 +0x206
  main.main()
      google.golang.org/grpc/_test/_testmain.go:218 +0x1d3
Goroutine 201 (finished) created at:
  google.golang.org/grpc.(*addrConn).connect()
      /home/travis/gopath/src/google.golang.org/grpc/clientconn.go:812 +0x185
  google.golang.org/grpc.(*acBalancerWrapper).Connect()
      /home/travis/gopath/src/google.golang.org/grpc/balancer_conn_wrappers.go:289 +0x88
  google.golang.org/grpc.(*pickfirstBalancer).HandleResolvedAddrs()
      /home/travis/gopath/src/google.golang.org/grpc/pickfirst.go:62 +0x4ab
  google.golang.org/grpc.(*ccBalancerWrapper).watcher()
      /home/travis/gopath/src/google.golang.org/grpc/balancer_conn_wrappers.go:139 +0x4e6
==================
--- FAIL: TestCloseConnectionWhenServerPrefaceNotReceived (3.50s)
	testing.go:699: race detected during execution of test
E1218 22:41:14.576531    5280 pickfirst.go:58] pickfirstBalancer: failed to NewSubConn: grpc: cannot create SubConn with empty address list
FAIL
FAIL	google.golang.org/grpc	45.236s
```

This was on a Go1.9 build.

--- a/clientconn_test.go
+++ b/clientconn_test.go
@@ -248,6 +248,21 @@ func TestCloseConnectionWhenServerPrefaceNotReceived(t *testing.T) {
 		t.Fatalf("Error while dialing. Err: %v", err)
 	}
 	<-done
+	// TODO: The code from BEGIN to END should be delete once issue
+	// https://github.com/grpc/grpc-go/issues/1750 is fixed.
+	// BEGIN
+	// Set underlying addrConns state to Shutdown so that no reconnect
+	// attempts take place and thereby resetting minConnectTimeout is
+	// race free.
+	client.mu.Lock()
+	addrConns := client.conns
+	client.mu.Unlock()
+	for ac := range addrConns {
+		ac.mu.Lock()
+		ac.state = connectivity.Shutdown
+		ac.mu.Unlock()
+	}
+	// END
 	client.Close()
 	close(clientDone)
 }

